core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -303,6 +303,9 @@ private TableProperties() {}
   public static final String ENGINE_HIVE_ENABLED = "engine.hive.enabled";
   public static final boolean ENGINE_HIVE_ENABLED_DEFAULT = false;
 
+  public static final String HIVE_LOCK_ENABLED = "hive.lock.enabled";
+  public static final boolean HIVE_LOCK_ENABLED_DEFAULT = true;
```

Comment:
Could we keep the `engine.hive` prefix? How about `engine.hive.lock-enabled`?

Reply:
Done

Reply:
Interesting, I  always read  'engine' as for hive-mr, ie running hive-on-iceberg, whereas this is for all engines (spark/flink) that use hive catalog.